plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/RefreshVersionsCleanupTask.kt

```diff
@@ -53,7 +53,10 @@ open class RefreshVersionsCleanupTask : DefaultTask() {
 
     @TaskAction
     fun cleanUpVersionsCatalog() {
-        if (VersionsCatalogs.isSupported() && FeatureFlag.VERSIONS_CATALOG.isEnabled) {
+        val settingsFiles = OutputFile.settingsFiles
+            .filter { it.existed }
+
+        if (settingsFiles.contains(OutputFile.GRADLE_VERSIONS_CATALOG) && VersionsCatalogs.isSupported() && FeatureFlag.VERSIONS_CATALOG.isEnabled) {
```

Comment:
The `FeatureFlag.VERSIONS_CATALOG.isEnabled` condition can be removed from the current code base. However, it is also a good idea to save it for future changes.

Reply:
We have it because we want users to be able to disable it in case there's a problem, without us having to rush a release to save the day.

We are keeping it until we're certain we have something super reliable (we are close to that, but better safe than sorry).